packages/algob/src/lib/algo-operator.ts

```diff
@@ -344,7 +354,73 @@ export class AlgoOperatorImpl implements AlgoOperator {
       creator: flags.sender.addr,
       txId: txInfo.txId,
       confirmedRound: confirmedTxInfo[confirmedRound],
-      appID: appId
+      appID: appId,
+      timestamp: Math.round(+new Date() / 1000)
+    };
+  }
+
+  /**
+   * Update programs for a contract.
+   * @param sender Account from which call needs to be made
+   * @param payFlags Transaction Flags
+   * @param appId ID of the application being configured or empty if creating
+   * @param newApprovalProgram New Approval Program filename
+   * @param newClearProgram New Clear Program filename
+   * @param flags Optional parameters to SSC (accounts, args..)
+   */
+  async updateSSC (
+    sender: algosdk.Account,
+    payFlags: rtypes.TxParams,
+    appID: number,
+    newApprovalProgram: string,
+    newClearProgram: string,
+    flags: rtypes.SSCOptionalFlags,
+    txWriter: txWriter
+  ): Promise<SSCInfo> {
+    const params = await tx.mkTxParams(this.algodClient, payFlags);
+
+    const app = await this.ensureCompiled(newApprovalProgram, false);
+    const approvalProg = new Uint8Array(Buffer.from(app.compiled, "base64"));
+    const clear = await this.ensureCompiled(newClearProgram, false);
+    const clearProg = new Uint8Array(Buffer.from(clear.compiled, "base64"));
+
+    const execParam: ExecParams = {
+      type: rtypes.TransactionType.UpdateSSC,
+      sign: rtypes.SignType.SecretKey,
+      fromAccount: sender,
+      appID: appID,
+      newApprovalProgram: newApprovalProgram,
+      newClearProgram: newClearProgram,
+      approvalProg: approvalProg,
+      clearProg: clearProg,
+      payFlags: payFlags,
+      accounts: flags.accounts,
+      foreignApps: flags.foreignApps,
+      foreignAssets: flags.foreignAssets,
+      appArgs: flags.appArgs,
+      note: flags.note,
+      lease: flags.lease
+    };
+
+    const txn = mkTransaction(execParam, params);
+    const txId = txn.txID().toString();
+    const signedTxn = txn.signTxn(sender.sk);
+
+    const txInfo = await this.algodClient.sendRawTransaction(signedTxn).do();
+    const confirmedTxInfo = await this.waitForConfirmation(txId);
+
+    const appId = confirmedTxInfo['application-index'];
+    const message = `Signed transaction with txID: ${txId}\nUpdated app-id: ${appId}`; // eslint-disable-line @typescript-eslint/restrict-template-expressions
```

Comment:
```suggestion
    const message = `Signed transaction with txID: ${txId}\nUpdated app: ${appId}`; // eslint-disable-line @typescript-eslint/restrict-template-expressions
```

Reply:
fair enough. suggested because it seemed like we updated application index (which we didn't). That's why renamed `app-id` to `app`